atomic_reactor/plugins/build_orchestrate_build.py

```diff
@@ -231,40 +272,38 @@ def get_cluster_info(self, cluster, platform):
         osbs = OSBS(conf, conf)
         try:
             current_builds = self.get_current_builds(osbs)
-        except OsbsException as e:
+        except OsbsException:
             # If the build is canceled reraise the error
-            if isinstance(e.cause, BuildCanceledException):
-                raise e
-
-            self.log.exception("Error occurred while listing builds on %s",
-                               cluster.name)
-            return ClusterInfo(cluster, platform, osbs, self.UNREACHABLE_CLUSTER_LOAD)
+            raise
 
         load = current_builds / cluster.max_concurrent_builds
         self.log.debug('enabled cluster %s for platform %s has load %s and active builds %s/%s',
                        cluster.name, platform, load, current_builds, cluster.max_concurrent_builds)
         return ClusterInfo(cluster, platform, osbs, load)
 
-    def choose_cluster(self, platform):
-        config = get_config(self.workflow)
-        clusters = [self.get_cluster_info(cluster, platform) for cluster in
-                    config.get_enabled_clusters_for_platform(platform)]
-
-        if not clusters:
-            raise RuntimeError('No clusters found for platform {}!'
-                               .format(platform))
+    def get_clusters(self, all_clusters, platform, retry_info):
+        ''' return clusters sorted by load '''
 
-        reachable_clusters = [cluster for cluster in clusters
-                              if cluster.load != self.UNREACHABLE_CLUSTER_LOAD]
+        clusters = []
+        while all_clusters and not clusters:
+            for cluster in all_clusters:
+                if retry_info.in_retry_wait(cluster):
+                    continue
+                try:
+                    clusters.append(self.get_cluster_info(cluster, platform))
+                except BuildCanceledException:
+                    raise
+                except OsbsException:
+                    if not retry_info.try_again_later(cluster, self.find_cluster_retry_delay):
+                        all_clusters.remove(cluster)
```

Comment:
Isn't this going to cause issues because you're altering the list being iterated over?